storage/storage_test.go

```diff
@@ -0,0 +1,238 @@
+package storage
+
+import (
+	"github.com/stretchr/testify/require"
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/iorhachovyevhen/dsss/models"
+)
+
+func TestNewStorageWithOptions(t *testing.T) {
+	tmpDir, err := ioutil.TempDir("", "badger")
+	require.Nil(t, err, err)
+
+	defer os.RemoveAll(tmpDir)
+
+	tmpFile := filepath.Join(tmpDir, "test")
+	opt := NewOptions().
+		WithDir(tmpFile).
+		WithValueDir(tmpFile).
+		WithValueLogFileSize(2 << 20)
+
+	storage := NewStorageWithOptions(opt)
+	defer storage.Close()
+	require.NotNil(t, storage)
+}
+
+func TestStorage_Add(t *testing.T) {
+	storage := NewDefaultStorage("/tmp/badger")
+	defer storage.Close()
+
+	data := models.NewSimpleData(
+		models.MetaData{
+			Title:    "test",
+			DataType: models.Simple,
+		},
+		randomContent(),
+	)
+
+	obtainedKey, err := storage.Add(data)
+	assert.Nil(t, err, err)
+	assert.Equal(t, data.ID(), obtainedKey)
+}
+
+func TestStorage_Read(t *testing.T) {
+	tmpDir, err := ioutil.TempDir("", "badger")
```

Comment:
All tests have the same begin and end. Maybe will be better to have one test with several sub tests?